### PR TITLE
Add `4.0.0-preview2` news

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
Screenshot 2024-09-26 at 08 48 34

Screenshot 2024-09-26 at 08 49 35

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @cloud-fan , @HyukjinKwon , @gengliangwang , @yaooqinn , @viirya , @huaxingao .

PyPi community didn't respond at all for one week. As a release manager, I believe we should not be blocked by the 3rd -party distribution channel. I'm going to release and announce it Today.

- https://lists.apache.org/thread/l1dxj201yt5thny5bnvckh7tzr7rcmxq
    - https://github.com/pypi/support/issues/4791

FYI, `pyspark-connect` is published. Only `pyspark` package is not published.
- https://pypi.org/project/pyspark-connect/#history

File: site/404.html
Comment (viirya) [inline site/404.html:182]: The link is correct but the link text "Preview release of Spark 4.0" is the same as preview 1. Do you think we can make them different?

**Review by viirya — APPROVED** [review]

File: site/404.html
Comment (dongjoon-hyun) [reply site/404.html:182]: Ya, I also felt in that way, but simply followed our convention at 3.0.0-preview2 here.
Screenshot 2024-09-26 at 09 23 48

Comment (dongjoon-hyun) [reply site/404.html:182]: It seems that we kept `New title itself` in the same way.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @viirya .

**Comment by dongjoon-hyun** [comment]
Let me merge this since it's already overdue.
- [[VOTE][RESULT] Release Spark 4.0.0-preview2 (RC1) - Friday, September 20](https://lists.apache.org/thread/4ctj2mlgs4q2yb4hdw2jy4z34p5yw2b1)

**Comment by cloud-fan** [comment]
late LGTM

File: site/404.html
Comment (viirya) [reply site/404.html:182]: Okay.

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan .

**Comment by yaooqinn** [comment]
The docker official images haven't been release yet, I have proposed https://github.com/docker-library/official-images/pull/17622 to fix

**Comment by dongjoon-hyun** [comment]
Thank you, @yaooqinn .

**Pull request merged**
